org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIDebugTarget.java

```diff
@@ -1754,8 +1755,9 @@ protected void resumeThreads() throws DebugException {
 	 *            the breakpoint that caused the suspension
 	 */
 	public void prepareToSuspendByBreakpoint(JavaBreakpoint breakpoint) {
-		setSuspended(true);
-		suspendThreads();
+		if (!fSuspended.getAndSet(true)) {
```

Comment:
I assume this will not be enough, but I'm running the reproducer test just to be sure.